includes/multi-currency/client/settings/single-currency/currency-preview.js

```diff
@@ -20,86 +20,62 @@ const CurrencyPreview = ( {
 	charmValue,
 } ) => {
 	const [ baseValue, setBaseValue ] = useState( 20.0 );
-	const [ calculatedValue, setCalculatedValue ] = useState( 0 );
 
-	const calculateCurrencyConversion = useCallback(
-		( value ) => {
-			const amount = parseFloat( value.toString().replace( /,/g, '.' ) );
-			const converted =
-				amount *
-				parseFloat( currencyRate ? currencyRate : targetCurrency.rate );
-			const rounded = parseFloat( roundingValue )
-				? Math.ceil( converted / parseFloat( roundingValue ) ) *
-				  parseFloat( roundingValue )
-				: converted;
-			const charmed = rounded + parseFloat( charmValue );
-			return isNaN( charmed )
-				? __( 'Please enter a valid number', 'woocommerce-payments' )
-				: formatCurrency(
-						isZeroDecimalCurrency( targetCurrency.code )
-							? charmed
-							: charmed * 100,
-						targetCurrency.code,
-						null,
-						true
-				  );
-		},
-		[ charmValue, currencyRate, roundingValue, targetCurrency ]
-	);
-
-	useEffect( () => {
-		if ( targetCurrency ) {
-			const initialCalculation = calculateCurrencyConversion( baseValue );
-			setCalculatedValue( initialCalculation );
-		}
-	}, [
-		calculateCurrencyConversion,
-		baseValue,
-		targetCurrency,
-		roundingValue,
-		charmValue,
-		currencyRate,
-	] );
-
-	const handleTextControlChange = ( value ) => {
-		setBaseValue( value );
-		const calculatedNewValue = calculateCurrencyConversion( value );
-		setCalculatedValue( calculatedNewValue );
-	};
+	const calculatedValue = useMemo( () => {
+		const amount = parseFloat( baseValue.toString().replace( /,/g, '.' ) );
+		const converted =
+			amount *
+			parseFloat( currencyRate ? currencyRate : targetCurrency.rate );
+		const rounded = parseFloat( roundingValue )
+			? Math.ceil( converted / parseFloat( roundingValue ) ) *
+			  parseFloat( roundingValue )
+			: converted;
+		const charmed = rounded + parseFloat( charmValue );
+		return isNaN( charmed )
+			? __( 'Please enter a valid number', 'woocommerce-payments' )
+			: formatCurrency(
+					isZeroDecimalCurrency( targetCurrency.code )
+						? charmed
+						: charmed * 100,
+					targetCurrency.code,
+					null,
+					true
+			  );
+	}, [ baseValue, charmValue, currencyRate, roundingValue, targetCurrency ] );
 
 	return (
-		<div className={ 'single-currency-settings-preview-wrapper' }>
-			<Card>
-				<CardBody>
-					<div>
-						<h4>{ storeCurrency.name }</h4>
```

Comment:
These `h4`s were used as labels for the input - updating them to be actual labels